### PR TITLE
Handle web layer offline

### PR DESCRIPTION
**Pull request by andrewazores**
SSIA. Addresses Issue #20. The server node application handles the web-layer connection failure by logging it and propagating a status 500 to the client application's request. Then the client simply shows "Update failed" type alerts at this point and controllers cancel future updates (this is not new to this PR).

### CODE REVIEW

File: server/public/js/common/application-service.js [hunk 1]
```diff
@@ -18,20 +18,17 @@ export default class ApplicationService {
...
-        window.alert(createPrefixedString(msg));
+        window.alert('[' + this.prefix + '] ' + msg);
     }
 
     log(msg) {
-        console.log(createPrefixedString(msg));
+        console.log('[' + this.prefix + '] ' + msg);
     }
 
-    createPrefixedString(str) {
```
Comment (jiekang) [inline server/public/js/common/application-service.js:32]: For what it's worth my fix was to call:

this.createPrefixedString(msg)

Which do you prefer? I kind of liked the function but to be honest it's such a small detail, it doesn't matter.

**Comment by andrewazores** [comment]
Yea, either way. I figured there doesn't seem to be much reason to expose that function to everyone who has an ApplicationService instance, so I got rid of it at the same time. I wouldn't be opposed to adding it back though.

File: server/index.js [hunk 2]
```diff
@@ -21,14 +21,13 @@ app.get('/:coll/latest', (req, res) => {
     hostname: '127.0.0.1',
     port: '8080',
     path: '/' + req.params.coll + '/latest',
-    method: 'GET'
   }
-  http.request(options, cpuRes => {
```
Comment (jiekang) [inline server/index.js:26]: I also had a fix here to handle the situation where the http response is chunked (returned over multiple calls for serverRes.on('data') in this case).

app.get(['/plugins/*', '/api/*'], (req, res) => {
  var options = {
    hostname: '127.0.0.1',
    port: '8090',
    path: req.originalUrl,
    method: 'GET'
  }

  http.request(options, webRes => {
    webRes.setEncoding('utf8');

    let rawData = '';

    webRes.on('data', chunk => {
      rawData += chunk;
    });

    return webRes.on('end', ()=> {
        res.send(rawData);
    });
  }).end();

});


Basically adds data continuously and then on 'end', sends it.

Can you incorporate this change as well?

I can merge after.

**Review by jiekang — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 2 above)
Comment (andrewazores) [reply server/index.js:26]: Thanks, I've just added that too.

**Review by jiekang — APPROVED** [review]
